### PR TITLE
update/typescript-layout

### PR DESCRIPTION
**Pull request by aemmadi**
Refactored `Layout/` from javascript to typescript

### CODE REVIEW

File: src/Components/Layout/NavBar.tsx [hunk 1]
```diff
@@ -9,16 +9,29 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 
 // import { Avatar, deepOrange } from "@material-ui/core"; (later use)
 
+interface NavBarProps {
+  dashboard: boolean;
+  signOut(): void;
+}
+
+interface NavBarState {
+  Navlink: any[];
```
Comment (harshasrikara) [inline src/Components/Layout/NavBar.tsx:18]: we can easily set a type for the Navlinks

type navlink {
     name: string,
     link: string,
}

and then we can use that

Navlink: navlink[],

File: src/Components/Layout/WorkshopEditor.tsx [hunk 2]
```diff
@@ -10,10 +10,23 @@ import draftToHtml from "draftjs-to-html";
 import "react-draft-wysiwyg/dist/react-draft-wysiwyg.css";
 import { Button } from "@material-ui/core";
 
-class WorkshopEditor extends React.Component {
-  constructor(props) {
+interface WorkshopEditorProps {
+  closeWindow(newText: string): void;
+  content: any;
```
Comment (harshasrikara) [inline src/Components/Layout/WorkshopEditor.tsx:15]: Is content not a string?

File: src/Components/Layout/WorkshopEditor.tsx [hunk 3]
```diff
@@ -10,10 +10,23 @@ import draftToHtml from "draftjs-to-html";
...
+}
+
+interface WorkshopEditorState {
+  editorState: any;
```
Comment (harshasrikara) [inline src/Components/Layout/WorkshopEditor.tsx:19]: VScode should give an inferred type for these two variables, recommend using them

File: src/react-app-env.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (harshasrikara) [inline src/react-app-env.d.ts:1]: What is this file?

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (harshasrikara) [inline tsconfig.json:1]: Use the same tsconfig settings as the other branches

Things like target are configured to be different --> es6 vs es5
Recommend just copying over the config and adding to it if required
Setting the branch to be downstream from `update/typescript` may help

File: src/Components/Layout/WorkshopEditor.tsx (hunk 2 above)
Comment (sivampatel) [reply src/Components/Layout/WorkshopEditor.tsx:15]: content should be HTML in string form

File: src/Components/Layout/WorkshopEditor.tsx (hunk 3 above)
Comment (sivampatel) [reply src/Components/Layout/WorkshopEditor.tsx:19]: I believe theres a editor state type definition if you look for draft js types.

File: src/react-app-env.d.ts (hunk 4 above)
Comment (aemmadi) [reply src/react-app-env.d.ts:1]: This was auto-generated

File: src/Components/Layout/ConfirmationDialog.tsx [hunk 6]
```diff
@@ -20,7 +20,7 @@ interface ConfirmationDialogProps {
 /**
  * This component represents a modal which pops up when needing to confirm deletion, no other button uses this component yet
  */
-class ConfirmationDialog extends React.Component<ConfirmationDialogProps> {
+class ConfirmationDialog extends React.Component<any, ConfirmationDialogProps> {
```
Comment (harshasrikara) [inline src/Components/Layout/ConfirmationDialog.tsx:23]: Should be `<ConfirmationDialogProps, any>`
props go first

**Comment by sivampatel** [comment]
Fixed marge conflicts. Basically merged current update/typescript-user branch into this branch.

**Pull request merged**